### PR TITLE
add support to servers with cryptkey in xkore 3

### PR DESCRIPTION
**Pull request by alisonrag**
add support to servers with cryptkey in xkore 3

xkore 3 in bRO:
xkore bro

xkore 3 in iRO:
xkore iro

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -55,3 +55,4 @@ tables/iRO/Restart/skillnametable-custom.txt
 tables/**/portalsLOS.txt
 tables/**/npcs.txt
 tables/**/monsters.txt
+*.xpm
```
Comment (lututui) [inline .gitignore]: What's this?

**Comment by lututui** [comment]
I can't walk in bRO: https://gist.githubusercontent.com/lututui/8cac8d7e7a98cf97a34e48d0352e6fb8/raw/f18ecb86f05eef6e4c5edf0d87424f858d4193f4/unable%2520to%2520move

File: src/Network/Receive/ServerType0.pm [hunk 2]
```diff
@@ -2801,6 +2802,8 @@ sub private_message_sent {
 sub sync_received_characters {
 	my ($self, $args) = @_;
 
+	return unless (UNIVERSAL::isa($net, 'Network::DirectConnection'));
```
Comment (lututui) [inline src/Network/Receive/ServerType0.pm]: Why?

Comment (alisonrag) [reply src/Network/Receive/ServerType0.pm]: If we permit openkore use this sub we will get duplicated sync in character server, after this we get dc

File: .gitignore (hunk 1 above)
Comment (alisonrag) [reply .gitignore]: My fault, but this is a xpm file auto generated by map.pl
Will be veru usefull

File: src/Network/XKoreProxy.pm [hunk 3]
```diff
@@ -79,15 +81,14 @@ sub new {
 		$self->{packetPending} = '';
 		$clientBuffer = '';
 	}
-
+	
+	$packetParser = Network::Receive->create($self, $masterServer->{serverType});
```
Comment (lututui) [inline src/Network/XKoreProxy.pm]: This should be done in eval { } so we can catch Exception::Class::Base (the same way it's done in sub checkServer)

File: src/Network/XKoreProxy.pm [hunk 4]
```diff
@@ -346,10 +367,6 @@ sub checkServer {
 			$self->{nextIp} = $master->{ip};
 			$self->{nextPort} = $master->{port};
 			message TF("Proxying to [%s]\n", $config{master}), "connection" unless ($self->{gotError});
-			eval {
-				$packetParser = Network::Receive->create($self, $masterServer->{serverType});
-				$messageSender = Network::Send->create($self, $masterServer->{serverType});
-			};
 			if (my $e = caught('Exception::Class::Base')) {
```
Comment (lututui) [inline src/Network/XKoreProxy.pm]: We're never catching an exception here

**Comment by moneyisgod** [comment]
Up

**Comment by Zexeta** [comment]
Does this actually work for anyone or am i the only one having 0 luck with it on Re:start?

**Pull request closed without merging**